core/src/test/kotlin/org/stellar/anchor/sep31/Sep31ServiceTest.kt

```diff
@@ -715,35 +814,138 @@ class Sep31ServiceTest {
+        Sep31PostTransactionResponse.builder()
+            .id(gotResponse!!.id)
+            .stellarAccountId("GA7FYRB5VREZKOBIIKHG5AVTPFGWUBPOBF7LTYG4GTMFVIOOD2DWAL7I")
+            .stellarMemo("")
+            .stellarMemoType("")
+            .build()
     assertEquals(wantResponse, gotResponse)
   }
+
+  @Test
+  fun test_info_response() {
+    val info = sep31Service.info
+    val jpyc = info.receive.get("JPYC")!!
+    val usdc = info.receive.get("USDC")!!
+
+    // Test correctness of reading test_assets.json
+    assertTrue(jpyc.enabled)
+    assertTrue(jpyc.quotesSupported)
+    assertTrue(jpyc.quotesRequired)
+    assertEquals(0, jpyc.feeFixed)
+    assertEquals(0, jpyc.feePercent)
+    assertEquals(1, jpyc.minAmount)
+    assertEquals(1000000, jpyc.maxAmount)
+    assertNotNull(jpyc.sep12)
+    assertNotNull(jpyc.sep12.sender)
+    assertNotNull(jpyc.sep12.sender.types)
+    assertEquals(1, jpyc.sep12.sender.types.size)
+    assertNotNull(jpyc.sep12.receiver)
+    assertNotNull(jpyc.sep12.receiver.types)
+    assertEquals(1, jpyc.sep12.receiver.types.size)
+    assertNotNull(jpyc.fields.transaction)
+    assertEquals(3, jpyc.fields.transaction.size)
+
+    assertTrue(usdc.enabled)
+    assertTrue(usdc.quotesSupported)
+    assertTrue(usdc.quotesRequired)
+    assertEquals(0, usdc.feeFixed)
+    assertEquals(0, usdc.feePercent)
+    assertEquals(1, usdc.minAmount)
+    assertEquals(1000000, usdc.maxAmount)
+    assertNotNull(usdc.sep12)
+    assertNotNull(usdc.sep12.sender)
+    assertNotNull(usdc.sep12.sender.types)
+    assertEquals(3, usdc.sep12.sender.types.size)
+    assertNotNull(usdc.sep12.receiver)
+    assertNotNull(usdc.sep12.receiver.types)
+    assertEquals(2, usdc.sep12.receiver.types.size)
+    assertNotNull(usdc.fields.transaction)
+    assertEquals(3, usdc.fields.transaction.size)
+
+    print(info)
+  }
+
+  @Test
+  fun test_validateRequiredFields() {
+    val assetInfo = assetService.getAsset("USDC")
+    Context.get().setAsset(assetInfo)
+    Context.get().setTransactionFields(txn.fields)
+    sep31Service.validateRequiredFields()
+
+    assetInfo.code = "BAD"
+    val ex1 = assertThrows<BadRequestException> { sep31Service.validateRequiredFields() }
+    assertEquals("Asset [BAD] has no fields definition", ex1.message)
+
+    Context.get().setAsset(null)
+    val ex2 = assertThrows<BadRequestException> { sep31Service.validateRequiredFields() }
+    assertEquals("Missing asset information.", ex2.message)
+
+    Context.get().setTransactionFields(null)
+    val ex3 = assertThrows<BadRequestException> { sep31Service.validateRequiredFields() }
+    assertEquals("'fields' field must have one 'transaction' field", ex3.message)
+  }
+
+  @Test
+  fun test_updateFee() {
+    val jwtToken = TestHelper.createJwtToken()
+    Context.get().setRequest(request)
+    Context.get().setJwtToken(jwtToken)
+
+    // With quote
+    Context.get().setQuote(quote)
+    request.destinationAsset = "USDC"
+    sep31Service.updateFee()
```

Comment:
👀  It'd be nice to validate if the outcome of the `sep31Service.updateFee()` call is what we expect.

Reply:
Updated.